consent/api/v1/views.py

```diff
@@ -103,13 +103,31 @@ def get_required_query_params(self, request):
 
         """
         if request.method in self.QUERY_PARAM_METHODS:
-            username = request.query_params.get(self.REQUIRED_PARAM_USERNAME, '')
-            course_id = request.query_params.get(self.REQUIRED_PARAM_COURSE_ID, '')
-            enterprise_customer_uuid = request.query_params.get(self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER)
+            username = request.query_params.get(
+                self.REQUIRED_PARAM_USERNAME,
+                request.data.get(self.REQUIRED_PARAM_USERNAME, '')
+            )
+            course_id = request.query_params.get(
+                self.REQUIRED_PARAM_COURSE_ID,
+                request.data.get(self.REQUIRED_PARAM_COURSE_ID, '')
+            )
+            enterprise_customer_uuid = request.query_params.get(
+                self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER,
+                request.data.get(self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER)
+            )
         else:
-            username = request.data.get(self.REQUIRED_PARAM_USERNAME, '')
-            course_id = request.data.get(self.REQUIRED_PARAM_COURSE_ID, '')
-            enterprise_customer_uuid = request.data.get(self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER)
+            username = request.data.get(
+                self.REQUIRED_PARAM_USERNAME,
+                request.query_params.get(self.REQUIRED_PARAM_USERNAME, '')
+            )
+            course_id = request.data.get(
+                self.REQUIRED_PARAM_COURSE_ID,
+                request.query_params.get(self.REQUIRED_PARAM_COURSE_ID, '')
+            )
+            enterprise_customer_uuid = request.data.get(
+                self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER,
+                request.query_params.get(self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER)
+            )
```

Comment:
I'm not following what's going on here. Could you explain in words? Seems like a lot of duplication of code.